src/PPN-microbench/ops.cpp

```diff
@@ -131,14 +128,14 @@ void Ops::run() {
         {
             std::jthread threads[cpus];
             for (size_t k = 0; k < cpus; k++) {
-                threads[k] = std::jthread([this, t1, haha] { wrap(SIMD_FLOAT_MAX_FN, (SIMD_FLOAT_MAX_TYPE *)haha, 1); });
+                threads[k] = std::jthread([this, t1, ptr] { wrap(SIMD_FLOAT_MAX_FN, (SIMD_FLOAT_MAX_TYPE *)ptr, 1); });
                 pthread_setaffinity_np(threads[k].native_handle(), sizeof(cpu_set_t), &cpusets[k]);
             }
         }
         t2 = high_resolution_clock::now();
         results[5][j] = (t2 - t1).count();
     }
```

Comment:
This snippet looks very much like duplicated code to me.
This is what you should wrap into a function that takes a function pointer (there is `std::function` for that in C++) so that you can simplify the code.

You should not need the `wrap` function if you pass dedicated scalar and vector micro-benchmarking routines (as you had before).